src/Interpreters/ProcessList.h

```diff
@@ -195,7 +216,9 @@ class QueryStatus : public WithContext
 
     QueryStatusInfo getInfo(bool get_thread_list = false, bool get_profile_events = false, bool get_settings = false) const;
 
-    CancellationCode cancelQuery(bool kill);
+    CancellationCode cancelQuery();
+
+    void onStop();
```

Comment:
Maybe it worth to put a comments for new methods and members that had been introduced here? since the code became more complex now.